lib/utils.js

```diff
@@ -204,12 +204,37 @@ is_valid_year = function(year, minYear, maxYear) {
 Checks that the number representing the day of the month is within a valid range
 
 :param string day: day to be validated
+:param string month: Optional. Month to be validated.
+:param string year: Optional. Year to be validated.
 */
-is_valid_day_of_month = function(day) {
-    // check that it is a number and between 1 and 31
-    return this.check_valid_number(day)
-        && parseInt(day, 10) >= 1
-        && parseInt(day, 10) <= 31;
+is_valid_day_of_month = function(day, month, year) {
+    if (month && year) {
+        return this.is_valid_date(year + "-" + month + "-" + day, "YYYY-M-D");
+    } else if (month) {
+        if (this.check_valid_number(month) && this.check_valid_number(day)) {
+            switch (parseInt(month, 10)) {
+                case 2:
+                    return (parseInt(day, 10) >= 1 && parseInt(day, 10) <= 29);
+                case 1:
+                case 3:
+                case 5:
+                case 7:
+                case 8:
+                case 10:
+                case 12:
+                    return (parseInt(day, 10) >= 1 && parseInt(day, 10) <= 31);
+                default:
+                    return (parseInt(day, 10) >= 1 && parseInt(day, 10) <= 30);
```

Comment:
This is not a pretty way of doing it, though I guess it works. I would've just done
`return this.is_valid_date("2000-" + month + "-" + day, "YYYY-M-D");`
The year 2000 is a leap year (I think), so it should cover all the date possibilities, including Feb 29.